src/Mini-Project-UI.py

```diff
@@ -0,0 +1,198 @@
+import numpy as np
+import sys
+import pydicom
+from PySide6 import QtWidgets
+from PySide6.QtGui import QPixmap, QImage
+from PySide6.QtCore import QTimer
+from PIL import Image
+import os
+
+def dicom_image_opener(ds):
+    """Returns an image to be displayed by PySide6"""
+    picture = ds.pixel_array
+    picture = picture.astype(np.float32)
+    picture -= picture.min()
+    picture /= picture.max()
+    picture *= 255
+    picture = picture.astype(np.uint8)
+
+    image = Image.fromarray(picture)
+    image = image.convert("L")
+    qimage = QImage(image.tobytes(), image.width, image.height, QImage.Format_Grayscale8)
+
+    return qimage
+
+class MiniProjectUI(QtWidgets.QDialog):
+    """The class that contains the UI"""
+    number_rows_for_dic_button = 2
+
+    def __init__(self):
+        super().__init__()
+        self.setWindowTitle("Mini Project UI")
+        self.path = ""
+        self.directory_button_box()
+        main_layout = QtWidgets.QVBoxLayout()
+        self.image_label = QtWidgets.QLabel("No Image to Display")
+
+        main_layout.addWidget(self._grid_group_box)
+        main_layout.addWidget(self.image_label)
+
+        self.setLayout(main_layout)
+
+        QTimer.singleShot(100,self.check_saved_dir)
+
+
+    #popup to allow the user to open a diretory checks if the user want to open or close the program
+    def please_select_directory(self):
+        """Popup to allow the user to select a directory"""
+        pop_up = QtWidgets.QMessageBox()
+        pop_up.setIcon(QtWidgets.QMessageBox.Question)
+        pop_up.setText("Please Select A DICOM File to Open")
+        pop_up.setStandardButtons(QtWidgets.QMessageBox.Close | QtWidgets.QMessageBox.Open)
+        anw = pop_up.exec()
+
+        if anw == QtWidgets.QMessageBox.Close:
+            QtWidgets.QApplication.exit()
+        elif anw == QtWidgets.QMessageBox.Open:
+            dir_path, _ = QtWidgets.QFileDialog.getOpenFileName(self, "Select DICOM File", filter="DICOM Files (*.dcm)")
+            self.path = dir_path
+            self.open_dicom_file()
+
+
+    def file_cannot_be_opened_error(self):
+        error_message = QtWidgets.QMessageBox()
+        error_message.setIcon(QtWidgets.QMessageBox.Warning)
+        error_message.setText("Could not open file sorry, file is either corrupted or incomplete")
```

Comment:
Could not open file.
File is either corrupted or incomplete.

Now, what can the user do about this, or is there a way for them to use another tool to determine if in fact the file is corrupted or incomplete?  What does incomplete mean in this context?

In a commercial environment, one might say "Contact ACME corporation support for assistance" or
briefly explain what the user can do to help themselves.

In this circumstance, if you know what is missing, tell them.
If you think the file is corrupt, tell them that, and suggest using a 3rd party DICOM validation tool to check the file.

Reply:
I would only include an apology if the software has had an internal error (one of those "you shouldn't have gotten here, don't know what happened").